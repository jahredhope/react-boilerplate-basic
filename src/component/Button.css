.button {
  display: block;
  line-height: 40px;
  min-width: 80px;
  padding: 0 10px 0 10px;
  box-sizing: border-box;
  position: relative;
  border-radius: 5px;
  background-color: var(--color-main);
  border: none;
  font-size: inherit;
  cursor: pointer;
  opacity: 1;
  transition: opacity 0.3s;
}

.button:focus {
  opacity: .8;
  outline-width: 0;
}
.button:hover {
  opacity: .8;
}
.button:active {
  opacity: .6;
  transition: opacity 0s;
}

.button-primary {
  background-color: green;
}
