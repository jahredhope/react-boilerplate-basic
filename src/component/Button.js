import React, {PropTypes} from 'react';
import cn from 'classnames'

import styles from './Button.css'

const Button = ({type, children, className}) => (
      <button className={cn(styles.button, type && styles[`button-${type}`], className)} >
        {children}
      </button>
        );

Button.propTypes = {
  type: PropTypes.oneOf(['primary', 'danger']),
  children: PropTypes.string.isRequired,
  className: PropTypes.string
}

export default Button
