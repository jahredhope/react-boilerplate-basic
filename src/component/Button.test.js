// Link.react-test.js
import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

test('Button sets a base button class and passes through children', () => {
  const component = renderer.create(
    <Button>mock-child</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button passes through unknown props', () => {
  const component = renderer.create(
    <Button form={'mock-form'}>mock-child</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button sets class based on type', () => {
  const component = renderer.create(
    <Button type={'danger'}>mock-child</Button>
  );
  let tree = component.toJSON();
  expect(tree.props.className).toMatch(/(^|\s)mock-style-button-danger($|\s)/)
  expect(tree).toMatchSnapshot();
});
