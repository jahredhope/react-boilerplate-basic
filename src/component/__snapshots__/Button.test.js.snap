exports[`test Button passes through unknown props 1`] = `
<button
  className="mock-style-button"
  form="mock-form">
  mock-child
</button>
`;

exports[`test Button sets a base button class and passes through children 1`] = `
<button
  className="mock-style-button">
  mock-child
</button>
`;

exports[`test Button sets class based on type 1`] = `
<button
  className="mock-style-button mock-style-button-danger">
  mock-child
</button>
`;
